### ISSUE TITLE
Crystal issues with flycheck

### ISSUE THREAD

**Issue by PixeLInc** [hlissner/doom-emacs#710]
### Observed behavior
I am editing a crystal file over ssh on my pi using Crystal 0.24.2-3.
I open the file, try to save it (it lets me but errors), press ESC and it errors out saying
how 
```
Suspicious state from syntax checker crystal-build: Flycheck checker crystal-build returned non-zero exit code 1, but its output container no errors: Error: File: /ssh:user@<host>:/file_path/flycheck_filename.cr does not exist.

Try installing a more recent version of crystal-build, and please open a bug report if the issue persists in the latest release.
```

### Expected behavior
It should just.. like.. not error :+1:

### Steps to reproduce

1. ssh into a rpi (idk if this issue persists over other platforms)
2. open/create a crystal file
3. It should error out right away, or when you save/press ESC

### System information

<details>
<summary>Click to expand</summary>

- OS: gnu/linux (x86_64-pc-linux-gnu)
- Emacs: 26.1 (May 28, 2018)
- Doom: 2.0.9 (develop 0354edc3a668fdba8b2b1922bcc50cd193dcef5f)
- Graphic display: t (daemon: nil)
- System features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES THREADS LIBSYSTEMD LCMS2
- Details:
  ```elisp
  elc count: 0
  uname -a:  Linux PixeL-Antegros 4.16.13-2-ARCH #1 SMP PREEMPT Fri Jun 1 18:46:11 UTC 2018 x86_64 GNU/Linux
  modules:   (:feature (evil +everywhere) file-templates (lookup +devdocs +docsets) snippets spellcheck (syntax-checker +childframe) version-control workspaces :completion (ivy +fuzzy) :ui doom-dashboard doom-modeline doom-quit evil-goggles hl-todo nav-flash neotree (popup +all +defaults) vi-tilde-fringe window-select :emacs ediff electric eshell imenu term :tools magit pdf rotate-text tmux upload :lang crystal csharp data erlang elixir emacs-lisp go (haskell +intero) (java +meghanada) javascript lua markdown (org +attach +babel +capture +export +present) php purescript python rust sh swift web :app :config)
  packages:  (enh-ruby-mode rbenv rinari rubocop yard-mode)
  exec-path: (~/.gem/ruby/2.5.0/bin ~/.cargo/bin /usr/local/bin /usr/bin /bin /usr/local/sbin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.1/x86_64-pc-linux-gnu)
  ```
</details>

My apology if this is an issue on my end with something I've done, not too experienced w/ emacs yet.

**Comment by hlissner** [comment]
This should be fixed as of e05d7cf. Let me know if that isn't the case!

**Comment by PixeLInc** [comment]
It seems to be fixed on my end as well, thanks!
